FEATURE_DEMO.md

```
@@ -0,0 +1,145 @@
+- GitGenie detects `.git/MERGE_HEAD` file indicating active merge
+- Prioritizes `recover` and `undo` commands as most relevant
+- Shows context reason for each suggestion
+- Displays current repository state
+
+---
+
+## Demo 2: Staged Changes Detection
+
+### Setup  
+```bash
+$ echo "new feature" > feature.ts
```

markdownlint-cli2 (0.22.1)

[warning] Dollar signs used before commands without showing output

(MD014, commands-show-output)